src/AppBundle/Form/UserType.php

```diff
@@ -23,6 +25,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'second_options' => ['label' => 'Tapez le mot de passe à nouveau'],
             ])
             ->add('email', EmailType::class, ['label' => 'Adresse email'])
+            ->add('roles', ChoiceType::class, [
+                'multiple' => true,
+                'choices' => [
+                    "Utilisateur" => "ROLE_USER",
```

Comment:
Do not hard code this list of user roles. Inject it into your form ;)

Reply:
inject `security.role_hierarchy.roles`

Reply:
I tried this in UserType:

```
    public function __construct(RoleHierarchyInterface $roles)
    {
        $this->roles = $roles;
    }
```

but I can't access the content of $roles, everything is private without any getters.

I can just use `$roles->getReachableRoles(["ROLE_ADMIN"]);` but I have to hard code the role admin here, so it's not very different from the initial code.